packages/cubejs-questdb-driver/src/QuestQuery.ts

```diff
@@ -0,0 +1,144 @@
+  public newFilter(filter: any) {
+    return new QuestFilter(this, filter);
+  }
+
+  public newParamAllocator(): ParamAllocator {
+    return new QuestParamAllocator();
+  }
+
+  public concatStringsSql(strings: string[]): string {
+    return `concat(${strings.join(', ')})`;
+  }
+
+  public convertTz(field: string): string {
+    return `to_timezone(${field}, '${this.timezone}')`;
+  }
+
+  public timeStampCast(value: string) {
+    return value;
+  }
+
+  public dateTimeCast(value: string) {
+    return value;
+  }
+
+  public subtractInterval(date: string, interval: string): string {
+    const [number, type] = this.parseInterval(interval);
+    return `dateadd('${type}', ${-number}, ${date})`;
+  }
+
+  public addInterval(date: string, interval: string): string {
+    const [number, type] = this.parseInterval(interval);
+    return `dateadd('${type}', ${number}, ${date})`;
+  }
+
+  public unixTimestampSql(): string {
+    // QuestDB's now() function returns epoch timestamp with microsecond granularity.
+
+    return 'now() / 1000000';
+  }
+
+  public timeGroupedColumn(granularity: string, dimension: string): string {
+    const interval = GRANULARITY_TO_INTERVAL[granularity];
+    if (interval === undefined) {
+      throw new Error(`${granularity} granularity is not supported`);
+    }
+    return `timestamp_floor('${GRANULARITY_TO_INTERVAL[granularity]}', ${dimension})`;
+  }
+
+  public dimensionsJoinCondition(leftAlias: string, rightAlias: string): string {
+    const dimensionAliases = this.dimensionAliasNames();
+    if (!dimensionAliases.length) {
+      return '1 = 1';
+    }
+    return dimensionAliases
+      .map(alias => `(${leftAlias}.${alias} = ${rightAlias}.${alias} OR (${leftAlias}.${alias} = NULL AND ${rightAlias}.${alias} = NULL))`)
+      .join(' AND ');
+  }
+
+  public orderHashToString(hash: any): string | null {
+    // QuestDB has partial support for order by index column, so map these to the alias names.
+
+    if (!hash || !hash.id) {
+      return null;
+    }
+
+    const fieldAlias = this.getFieldAlias(hash.id);
+
+    if (fieldAlias === null) {
+      return null;
+    }
+
+    const direction = hash.desc ? 'DESC' : 'ASC';
+    return `${fieldAlias} ${direction}`;
+  }
+
+  private getFieldAlias(id: string): string | null {
+    const equalIgnoreCase = (a: any, b: any) => (
+      typeof a === 'string' && typeof b === 'string' && a.toUpperCase() === b.toUpperCase()
+    );
+
+    let field;
+
+    field = this.dimensionsForSelect().find(
+      (d: any) => equalIgnoreCase(d.dimension, id),
+    );
+
+    if (field) {
+      return field.aliasName();
+    }
+
+    field = this.measures.find(
+      (d: any) => equalIgnoreCase(d.measure, id) || equalIgnoreCase(d.expressionName, id),
+    );
+
+    if (field) {
+      return field.aliasName();
+    }
+
+    return null;
+  }
+
+  public groupByClause(): string {
+    // QuestDB doesn't support group by index column, so map these to the alias names.
```

Comment:
Can you please provide a brief example of what this is a workaround for? What kind of cube config / query combo leads to a group by index column being issued?

Reply:
Added an example in df4d6df2c84c483d0a6aa5d8c51a306b23d6e9ff

Reply:
Thanks!